### PR TITLE
Add values.py processor for environmental overrides of subcharts and mariadb chart refactor

### PR DESCRIPTION
**Pull request by alanmeadows**
This script allows you to have an single environmental YAML file that can be
chunked up for feeding into various subcharts.  This is required as values
input into helm requires the top level namespace be stripped off.  In other
words, a flat namespace.  This allows a single environmental definition that
can be chunked into YAML files for subchart input.

Usage will be documented on the wiki.

---
This change is [Reviewable](https://reviewable.io/reviews/att-comdev/aic-helm/49)

Also included in this pull request:

Mariadb chart refactor

This is a major refactor of the mariadb chart.  A few things
are accomplished:

* The chart template layout is updated to match our keystone
chart, providing a more structure to the chart.

* The chart was updated to leverage StatefulSets, which requires
Kubernetes 1.5, and helm 2.1.0.

* The bootstrapping process was completely overhauled to support
the unique constraints of statefulsets, namely that they come up
one by one, needing the previous to be in a ready state before
the next is provisioned.

* The references to {{ .IP }} we removed and replaced with POD_IP
environmental passing and address binding was fixed in several
places for wsrep functionality.  This may explain several oddities
with the previous setup causing mysterious and intermittent
database consistency issues.

### CODE REVIEW

File: common/utils/values/values.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    if item in yamldata.keys():
+                        f.write(yaml.dump({item: yamldata[item]}, default_flow_style=False))
+                f.write(yaml.dump(yamldata[top_level], default_flow_style=False))
+
+        else:
+            LOG.exception("Unable to create path %s based on top level key in %s" % (top_level_path, yaml_path))
+
+def run(args):
+
+    if args.yaml:
```
Comment (alanmeadows) [inline common/utils/values/values.py]: I wondered if someone was going to catch this ;-)

File: common/utils/values/values.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+def process_yaml(yaml_path):
+    '''
+    Processes YAML file and generates breakdown of top level
+    keys
+    '''
+
+    # perform some basic validation
+    if not os.path.isfile(yaml_path):
+        LOG.exception("The path %s does not point to a valid file", yaml_path)
+
```
Comment (alanmeadows) [inline common/utils/values/values.py]: exit is sufficient

File: common/utils/values/values.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    ap = argparse.ArgumentParser(description=DESCRIPTION)
+    ap.add_argument('-d', '--debug', action='store_true',
+                    default=False, help='Enable debugging')
+    ap.add_argument('-y', '--yaml', action='store',
+                    help='Path to Master YAML File')
+    return ap.parse_args()
+
+def setup_logging(args):
+    level = logging.INFO
```
Comment (alanmeadows) [inline common/utils/values/values.py]: agreed

**Comment by alanmeadows** [comment]
@DTadrzak @PiotrProkop accidently let the mariadb refactor slip into this existing pull request.  This could use a second set of eyes if you have a moment.  May be of interest as it is a different path then what was pulled in from Stackanetes.

**Comment by alanmeadows** [comment]
@DTadrzak feedback for values.py incorporated

**Comment by v1k0d3n** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 36 of 37 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/aic-helm/49#-:-KZPWRikgDl5rtme54zO:bnfp4nl)*

**Comment by v1k0d3n** [comment]
this closes #51

**Pull request merged**